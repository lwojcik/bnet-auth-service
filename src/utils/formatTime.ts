import { Duration } from 'luxon';

export const formatTime = (seconds: number) =>
  Duration.fromObject({ seconds }).toISOTime();
