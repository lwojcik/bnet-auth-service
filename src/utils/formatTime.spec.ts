import { formatTime } from './formatTime';

describe('formatTime', () => {
  const testTimes = [8, 16, 32, 64, 128, 32768, 128000];

  it('should format time correctly', () => {
    testTimes.forEach((testTime) => {
      expect(formatTime(testTime)).toMatchSnapshot();
    });
  });
});
