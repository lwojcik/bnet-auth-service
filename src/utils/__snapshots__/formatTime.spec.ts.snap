// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatTime should format time correctly 1`] = `"00:00:08.000"`;

exports[`formatTime should format time correctly 2`] = `"00:00:16.000"`;

exports[`formatTime should format time correctly 3`] = `"00:00:32.000"`;

exports[`formatTime should format time correctly 4`] = `"00:01:04.000"`;

exports[`formatTime should format time correctly 5`] = `"00:02:08.000"`;

exports[`formatTime should format time correctly 6`] = `"09:06:08.000"`;

exports[`formatTime should format time correctly 7`] = `null`;
